### PR TITLE
fix(linter): detect vitest jest alias rules

### LINKED ISSUE #7240: linter: `jest` and `vitest` compatible rules are  reported as not supported
### What version of Oxlint are you using?

0.11.1

### What command did you run?

npx oxlint -c .oxlint.json

### What does your `.oxlint.json` config file look like?

```
{
  "categories": {
    "correctness": "off"
  },
  "rules": {
    "no-identical-title": "error",
//    "vitest/no-identical-title": "error", -- also report jest/no-identical-title not supported
//    "jest/no-identical-title": "error"  -- also reports jest/no-identical-title not supported
  }
}

```

### What happened?

```
  ! The following rules do not match the currently supported rules:
  | jest/no-identical-title
```

The same does happend for other rules rules `jest/no-focused-tests`. I think all current rules are effected which should be the following:

```
  'consistent-test-it',
  'expect-expect',
  'no-alias-methods',
  'no-conditional-expect',
  'no-conditional-in-test',
  'no-commented-out-tests',
  'no-disabled-tests',
  'no-focused-tests',
  'no-identical-title',
  'no-restricted-jest-methods',
  'no-test-prefixes',
  'prefer-hooks-in-order',
  'valid-describe-callback',
  'valid-expect',
```

### PR DESCRIPTION
**Pull request by Sysix**
closes #7240

Can not find a better place / method to add a test case

### CODE REVIEW

File: npm/oxc-wasm/oxc_wasm_bg.wasm
Comment (camc314) [inline npm/oxc-wasm/oxc_wasm_bg.wasm:1]: @Boshen should this be .gitignored?

Comment (Sysix) [reply npm/oxc-wasm/oxc_wasm_bg.wasm:1]: im sorry. I failed my merge with main. Will delete it >.<

**Review by Boshen — CHANGES_REQUESTED** [review]
Merge conflict.

**Review by camchenry — COMMENTED** [review]
Merge conflict aside, I think this makes sense

File: crates/oxc_linter/src/builder.rs [hunk 1]
```diff
@@ -421,9 +421,16 @@ impl RulesCache {
         let mut all_rules: Vec<_> = if self.plugins.is_all() {
             RULES.clone()
         } else {
+            let mut plugins = self.plugins;
+
+            // we need to include some jest rules when vitest is enabled, see [`VITEST_COMPATIBLE_JEST_RULES`]
+            if plugins.contains(LintPlugins::VITEST) {
+                plugins = plugins.union(LintPlugins::JEST);
+            }
```
Comment (camc314) [inline crates/oxc_linter/src/builder.rs:429]: would this enable the default jest rules?

Comment (Sysix) [reply crates/oxc_linter/src/builder.rs:429]: I do not think. This is only the `RulesCache` where the Linter will pick available rules. 
Maybe someone with more knowledge can answer this :)

**Review by Boshen — APPROVED** [review]

**Pull request merged**